pkg/controller/machineset/machineset_controller_test.go

```diff
@@ -17,196 +17,134 @@ limitations under the License.
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-var c client.Client
+var _ = Describe("MachineSet Reconciler", func() {
+	namespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "ms-test"}}
 
-var expectedRequest = reconcile.Request{NamespacedName: types.NamespacedName{Name: "foo", Namespace: "default"}}
+	BeforeEach(func() {
+		By("Setting up a new manager")
+		mgr, err := manager.New(cfg, manager.Options{MetricsBindAddress: "0"})
+		Expect(err).NotTo(HaveOccurred())
 
-const timeout = time.Second * 5
+		k8sClient = mgr.GetClient()
 
-// TODO: drop this in favour of ginkgo like validations
-// similar to https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/machineset_controller_test.go
-func TestReconcile(t *testing.T) {
-	// Setup the Manager and Controller.  Wrap the Controller Reconcile function so it writes each request to a
-	// channel when it is finished.
-	mgr, err := manager.New(cfg, manager.Options{MetricsBindAddress: "0"})
-	if err != nil {
-		t.Errorf("error creating new manager: %v", err)
-	}
-	c = mgr.GetClient()
+		By("Setting up a new reconciler")
+		reconciler := newReconciler(mgr)
 
-	r := newReconciler(mgr)
-	recFn, requests := SetupTestReconcile(r)
-	if err := add(mgr, recFn, r.MachineToMachineSets); err != nil {
-		t.Errorf("error adding controller to manager: %v", err)
-	}
+		err = add(mgr, reconciler, reconciler.MachineToMachineSets)
+		Expect(err).NotTo(HaveOccurred())
 
-	stop, errChan := StartTestManager(mgr, t)
-	defer func() {
-		close(stop)
-		if err := <-errChan; err != nil {
-			t.Fatalf("error starting test manager: %v", err)
-		}
-	}()
-
-	replicas := int32(2)
-	labels := map[string]string{"foo": "bar"}
-
-	testCases := []struct {
-		name            string
-		instance        *machinev1beta1.MachineSet
-		expectedRequest reconcile.Request
-		verifyFnc       func()
-	}{
-		{
-			name: "Refuse invalid machineset (with invalid matching labels)",
-			instance: &machinev1beta1.MachineSet{
-				ObjectMeta: metav1.ObjectMeta{Name: "invalidfoo", Namespace: "default"},
-				Spec: machinev1beta1.MachineSetSpec{
-					Replicas: &replicas,
-					Selector: metav1.LabelSelector{
-						MatchLabels: map[string]string{"foo": "bar"},
-					},
-					Template: machinev1beta1.MachineTemplateSpec{
-						ObjectMeta: metav1.ObjectMeta{
-							Labels: map[string]string{"foo": "bar2"},
-						},
-					},
+		By("Starting the manager")
+		go func() {
+			defer GinkgoRecover()
+			Expect(mgr.Start(doneMgr)).To(Succeed())
+		}()
+
+		By("Creating the namespace")
+		Expect(k8sClient.Create(ctx, namespace)).To(Succeed())
+	})
+
+	AfterEach(func() {
+		By("Deleting the namespace")
+		Expect(k8sClient.Delete(ctx, namespace)).To(Succeed())
+
+		By("Deleting the machinesets")
+		cleanResources()
```

Comment:
I would swap the order of these around so you clean the resources before deleting the namespace